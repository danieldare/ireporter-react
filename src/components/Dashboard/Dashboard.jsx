import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

const dashboard = props => (
  <div className="container">
    <div className="title-user">
      <span className="user-info2">My Dashboard</span> <br />
      <small id="username">
        Welcome, {props.auth.isAuthenticated ? props.auth.user.email : 'guest'}
      </small>
    </div>
    <hr />
    <div className="main-section">
      <p className="left-head">
        Tell Us About it, <br />
        <span>We can build a better Nation</span>
      </p>
      <div>
        <div className="card">
          <div className="card-title">
            <p>Red Flag</p>
          </div>
          <div className="card-img" />
          <div className="card-body">
            <small className="bold">Description:</small>
            <p className="mt-1">
              You can tell us about an incident linked to corruption e.g bribery
              case, fraud. e.t.c
            </p>
            <Link to="/create-redflag" className="btnn btn-dark mt-1">
              Create redflag
            </Link>
          </div>
        </div>
        <Link to="/view-redflag" className="link btn-dark">
          View Red-flag Record
        </Link>
      </div>
      <div>
        <div className="card">
          <div className="card-title">
            <p>Intervention</p>
          </div>
          <div className="card-second-img" />
          <div className="card-body">
            <small className="bold">Description:</small>
            <p className="mt-1">
              A call for a government agency to intervene e.g repair bad road
              sections, collapsed bridges. e.t.c
            </p>
            <Link to="/create-intervention" className="btnn btn-brown mt-1">
              Create Intervention
            </Link>
          </div>
        </div>
        <Link to="/view-intervention" className="link2">
          View Intervention Record
        </Link>
      </div>
    </div>
  </div>
);

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(dashboard);
