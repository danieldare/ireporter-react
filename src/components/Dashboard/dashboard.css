/* DASHBOARD */
/* Card */
.card {
  border-radius: 2px;
  background-color: #fafafa;
  height: 400px;
  box-shadow: 0 4px 10px 0 rgba(0, 0, 0, 0.4);
  width: 380px;
  margin: 30px;
  overflow: hidden;
  transition: transform 1s ease-in-out;
  cursor: pointer;
  font-weight: 500;
  font-size: 4rem;
  font-family: 'Dosis', sans-serif;
}

.card:hover {
  box-shadow: 0 5px 10px 0 rgba(0, 0, 0, 0.5);
  transform: scale(1.02);
}

.card-title {
  position: relative;
  z-index: 3;
  top: 40%;
}

.card-title p {
  position: absolute;
  width: 100%;
  margin: 0 auto;
  text-align: center;
  color: #fff;
  line-height: 2rem;

  font-weight: 300;
  font-weight: 500;
  font-size: 3.7rem;
  font-family: 'Dosis', sans-serif;
}

.card-img {
  background-image: linear-gradient(
      rgba(31, 31, 31, 0.39),
      rgba(209, 209, 209, 0.3)
    ),
    url('https://res.cloudinary.com/decqfpglp/image/upload/v1551186082/bribe.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  width: 100%;
  height: 67%;
  overflow: hidden;
  display: block;
  opacity: 0.8;
}

.card-second-img {
  background-image: linear-gradient(
      rgba(219, 130, 103, 0.604),
      rgba(59, 59, 59, 0.08)
    ),
    url('https://res.cloudinary.com/decqfpglp/image/upload/v1551186082/bribe4.jpg');
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
  width: 100%;
  height: 67%;
  display: block;
  opacity: 0.9;
  margin: 0;
  padding: 0;
}

.card-img,
.card-second-img {
  transition: transform 1s;
  cursor: pointer;
}

.card-img:hover,
.card-second-img:hover {
  transform: scale(1.02);
}

.card-body {
  font-size: 1.5rem;
  padding: 10px;
}

.mt-1 {
  margin-top: 10px;
}
.mt-2 {
  margin-top: 20px;
}

.bold {
  font-weight: 600;
}

/* End of Card */
.flex-row {
  display: flex;
  justify-content: space-around;
  margin: 0;
  padding: 0;
}

.main-section {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  /* margin-top: %; */
  margin-bottom: 12%;
}

.user-info {
  padding: 12px;
  color: steelblue;
  border-radius: 2px;
  font-weight: 600;
  font-size: 3rem;
  display: inline-block;
  margin: 0;
  padding: 0;
  margin-top: 15px;
  text-transform: uppercase;
  letter-spacing: 0.6rem;
  font-family: 'Nunito', sans-serif;
}

.title-user {
  color: #fff;
}

.title-user small {
  font-size: 1.4rem;
  color: #425868;
  font-weight: 600;
}

.user-info2 {
  color: #db8367;
  padding: 12px;

  border-radius: 2px;
  font-weight: 500;
  font-size: 3rem;
  display: inline-block;
  margin: 0;
  padding: 0;
  margin-top: 15px;
  text-transform: capitalize;
  letter-spacing: 0.1rem;
}

#username {
  text-transform: initial;
  font-size: 1.5rem;
}

.left-head {
  font-size: 2.5rem;
  color: #485f72;
  font-weight: 500;
}

.left-head span {
  color: #db8367;
}

.title {
  color: grey;
  font-size: 18px;
}

.btnn {
  border: none;
  outline: 0;
  display: inline-block;
  padding: 8px;
  color: white;
  background-color: #425a6d;
  text-align: center;
  cursor: pointer;
  width: 100%;
  border-radius: 2px;
  text-decoration: none;
  z-index: 9999999;
}

.btn-brown {
  background: #db8367;
}

button:hover,
a:hover {
  opacity: 0.8;
}

.link {
  background: #425a6d;
  display: inline-block;
  position: relative;
  left: 31px;
  padding: 10px;
  width: 380px;
  color: #fff;
  text-align: center;
  text-decoration: none;
  border-radius: 5px;
}

.link2 {
  background: #db8367;
  display: inline-block;
  position: relative;
  left: 31px;
  padding: 10px;
  width: 380px;
  color: #fff;
  text-align: center;
  text-decoration: none;
  border-radius: 5px;
}

/* @media queries */
@media (max-width: 1290px) {
  .bb {
    font-size: 1.5rem;
    width: 50%;
    display: block;
    margin: 15px auto;
  }

  .left-head {
    font-size: 3rem;
    font-weight: 600;
    color: #485f72;
    text-align: center;
    width: 100%;
    position: relative;
    top: 35px;
  }

  .card {
    height: 400px;
    width: 360px;
    position: relative;
    top: 65px;
  }

  .link,
  .link2 {
    top: 65px;
    width: 360px;
  }

  .main-section {
    flex-wrap: wrap;
    padding-bottom: 136px;
  }
}

@media (max-width: 800px) {
  .main-section {
    flex-wrap: wrap;
    padding-bottom: 136px;
  }
  header {
    transition: all 300ms ease-in-out;
  }

  header ul {
    padding: 0;
    transition: all 300ms ease-in-out;
  }

  .nav-items {
    font-size: 1.6rem;
    width: 100%;
  }

  .show-items {
    display: none;
    transition: all 300ms ease-in-out;
  }

  .nav-items li {
    transition: all 300ms ease-in;
    text-align: center;
    display: block;
  }

  .nav-items a {
    display: block;
    padding: 0;
    transition: all 300ms ease-in-out;
    background-color: #384753;
    border-bottom: 1px solid rgb(255, 255, 255);
  }

  .bar {
    color: #fff;
    float: right;
    display: block;
  }
}

@media (max-width: 867px) {
  .left-head {
    font-size: 2.3rem;
  }

  .link2 {
    margin-bottom: 80px;
  }

  .main-section {
    flex-wrap: wrap;
    padding-bottom: 80px;
  }

  #bg-img {
    max-height: 100%;
    background-size: cover;
    background-repeat: repeat;
  }
}

@media (max-width: 480px) {
  .left-head {
    left: 0%;
  }

  .card {
    height: 448px;
    width: 300px;
    position: relative;
    top: 65px;
    margin: 30px auto;
  }

  .link,
  .link2 {
    top: 65px;
    left: 0;
    width: 300px;
  }

  .link2 {
    margin-bottom: 80px;
  }

  .main-section {
    display: block;
    margin: 0 auto;
    text-align: center;
  }

  #bg-img {
    max-height: 100%;
    background-size: cover;
    background-repeat: repeat;
  }
}

@media (max-width: 350px) {
  .left-head {
    left: 0;
    font-size: 1.7rem;
  }

  .link,
  .link2 {
    top: 65px;
    left: 0;
    width: 241px;
  }

  .card {
    height: 429px;
    width: 241px;
    position: relative;
  }
}

.bb {
  font-size: 1.5rem;

  /* width: 50%; */
  display: block;
  margin: 15px auto;
}
