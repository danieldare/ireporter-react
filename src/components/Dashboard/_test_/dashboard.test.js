import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow, mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import Dashboard from '../Dashboard';
import { mockStore } from '../../../../test/setupTests.js';

const props = {
  auth: {
    isAuthenticated: false,
    user: {
      email: 'abc@example.com'
    }
  }
};

describe('<Register/> rendering', () => {
  it('should render without crashing', () => {
    const store = mockStore(props);
    const wrapper = mount(
      <Provider store={store}>
        <Router>
          <Dashboard />
        </Router>
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
