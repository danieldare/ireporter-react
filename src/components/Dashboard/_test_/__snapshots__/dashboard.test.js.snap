// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register/> rendering should render without crashing 1`] = `ReactWrapper {}`;
